enhancements/local-storage/automatic-detection.md

```diff
@@ -0,0 +1,275 @@
+}
+
+type AutoDetectVolumeStatus struct {
+    Status string `json:"status"`
+}
+
+// DeviceMechanicalProperty holds the device's mechanical spec. It can be rotational, nonRotational or
+// RotationalAndNonRotational
+type DeviceMechanicalProperty string
+
+const (
+    // The mechanical properties of the devices
+    // Rotational refers to magnetic disks
+    Rotational DeviceMechanicalProperty = "Rotational"
+    // NonRotational refers to ssds
+    NonRotational DeviceMechanicalProperty = "NonRotational"
+    // RotationalAndNonRotational refers to both magnetic and ssds
+    RotationalAndNonRotational DeviceMechanicalProperty = ""
+)
+
+type DeviceDiscoveryPolicyType string
+
+const (
+    // The DeviceDiscoveryPolicies that will be supported by the LSO.
+    // These Discovery policies will based on lsblk's type output
+    // Disk represents a device-type of disk
+    Disk DeviceDiscoveryPolicyType = "disk"
+    // Part represents a device-type of partion
+    Part DeviceDiscoveryPolicyType = "part"
+)
+
+type DeviceInclusionSpec struct {
+    // DeviceTypes that should be used for automatic detection. This would be one of the types supported
+    // by the local-storage operator. Currently the supported types are: disk,part
+    // If the list is empty the default value will be `[disk]`.
+    DeviceTypes []DeviceDiscoveryPolicyType `json:"deviceType"`
+
+    // DeviceMechanicalProperty denotes whether Rotational or NonRotational disks should be used.
+    // by default, RotationalAndNonRotational which matches all disks.
+    // +optional
+    DeviceMechanicalProperty DeviceMechanicalProperty `json:"deviceMechanicalProperty"`
+
+    // MinSize is the minimum size of the device which needs to be included
+    // +optional
+    MinSize resource.Quantity `json:"minSize"`
+
+    // MaxSize is the maximum size of the device which needs to be included
+    // +optional
+    MaxSize resource.Quantity `json:"maxSize"`
+
+    // Models is a list of device models. If not empty, the device's model as outputted by lsblk needs
+    // to contain at least one of these strings.
+    // +optional
+    Models []string `json:"models"`
+
+    // Vendors is a list of device vendors. If not empty, the device's model as outputted by lsblk needs
+    // to contain at least one of these strings.
+    // +optional
+    Vendors []string `json:"vendors"`
+}
+```
+
+The existing LSO daemon will create PVs that match the `AutoDetectVolume` criteria.
+
+### Test Plan
+
+- The integration tests for the LSO already exist. These tests will need to be updated to test this feature.
+- The tests must ensure that detection of devices are working/updating correctly.
+- The tests must ensure that data corruption are not happening during auto detection of devices.
+
+### Graduation Criteria
+
+- Documentation exists for the behaviour of each configuration item.
+- Unit and End to End tests coverage is sufficient.
+
+#### Examples
+
```

Comment:
At this point - we should also start designing the flow of device discovery from control-plane and into the node.

Reply:
How about we proceed with this design PR and save device discovery and UI flow for a future PR? Seems like they are fully independent per my comment above about not having a dry-run flag.

Reply:
makes sense to have a separate PR for the flow.

Reply:
By future PR - do you mean another design PR? Do we expect that to merge by enhancement freeze deadline?  The reason I have been asking for flow is because - I am not sure how this new type will be used. Are we going to modify existing controllers to watch `AutoDetectVolume` types too? It will be good to have some clarity before we can be sure about the design.